src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -193,6 +201,67 @@ public async Task SslStream_NestedAuth_Throws()
             }
         }
 
+        [Fact]
+        public async Task SslStream_UntrustedCaWithCustomCallback_OK()
+        {
+            var options = new  SslClientAuthenticationOptions() { TargetHost = "localhost" };
+            options.RemoteCertificateValidationCallback =
+                (sender, certificate, chain, sslPolicyErrors) =>
+                {
+                    chain.ChainPolicy.ExtraStore.AddRange(_serverChain);
```

Comment:
You probably want to still return false if sslPolicyErrors has either RemoteCertificateNameMismatch or RemoteCertificateNotAvailable set.  (Currently you'll get exceptions in the latter case, since certificate will be null, and chain might as well).